internal/server/data/identity.go

```diff
@@ -51,8 +51,8 @@ func AssignIdentityToGroups(tx GormTxn, user *models.Identity, provider *models.
 	}
 	var addIDs []idNamePair
 
-	stmt := `SELECT id, name FROM groups WHERE name IN (?)`
-	rows, err := tx.Query(stmt, groupsToBeAdded)
+	stmt := `SELECT id, name FROM groups WHERE name IN (?) AND organization_id = ?`
+	rows, err := tx.Query(stmt, groupsToBeAdded, user.OrganizationID)
```

Comment:
minor: should we use `tx.OrganizationID()` here, for consistent with other queries?

`user.OrganizationID` should be safe, even if it's not set, since the 0 value won't match anything.